.github/workflows/plan.yml

```diff
@@ -2,14 +2,35 @@ name: terraform
 on:
   pull_request:
     branches: [main]
+  pull_request_target:
+    branches: [main]
 
 jobs:
-  plan:
+  validate:
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      EMAIL: ${{ github.event.pusher.email }}
+      PR_REPO_FULL_NAME: ${{ github.event.pull_request.head.repo.full_name }}
     runs-on: ubuntu-latest
+    outputs:
+      valid_workflow: ${{ steps.validate_workflow.outputs.valid_workflow }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: validate_workflow
+        run: |
+          pip install PyGithub
```

Comment:
Could we use a lock file here? I think we'll be giving the `PyGithub` and dependencies access to the secret token so it'd be good to verify there's been no tampering.

Best lock file I know of is the Poetry one?